src/api/search/src/search.js

```diff
@@ -103,4 +101,57 @@ const search = async (options) => {
     values: hits.hits.map(({ _id }) => ({ id: _id, url: `${POSTS_URL}/${_id}` })),
   };
 };
-module.exports = { search };
+
+/**
+ * authorAutocomplete allows for querying author autocomplete fields
+ * @param { author } - text to search in author field
+ * @return all the results matching the fields text
+ */
+const authorAutocomplete = async ({ author }) => {
+  const results = {
+    query: {
+      match: {
+        'author.autocomplete': {
+          query: author,
+          operator: 'and',
+        },
+      },
+    },
+    highlight: {
+      fields: {
+        'author.autocomplete': {},
+      },
+    },
+  };
+
+  const {
+    body: { hits },
+  } = await client.search({
+    size: 10000,
+    _source: ['author'],
+    index,
+    body: results,
+  });
+
+  // Filter through all authors and remove duplicates
+  const authors = hits.hits.reduce(
+    (acc, { _source }) => (acc.includes(_source.author) ? acc : [...acc, _source.author]),
+    []
+  );
+
+  // Filter through all highlighted authors and remove duplicates
+  const highlights = hits.hits.reduce(
```

Comment:
Since highlight and author are from the same search result, we could merge them into one array instead of using the same index to get the highlight and author of a result. The object can look like this.
```js
[{
author: 'David Li',
highlight: '"<em>David</em> Li"
},
{...}
]
```

Reply:
Btw, should we enable fuzzy query in the `/autocomplete` route?

Reply:
That depends, how accurate we'd like to be with name searches, or how fuzzy we want it to be. 
I personally don't think we need it for just names.

Reply:
> Since highlight and author are from the same search result, we could merge them into one array instead of using the same index to get the highlight and author of a result. The object can look like this.
> 
> ```js
> [{
> author: 'David Li',
> highlight: '"<em>David</em> Li"
> },
> {...}
> ]
> ```

Yeah, agree to merge it, I think it would be easier in the front end to use this data if we want the highlight matching the author.